src/extension/app-schema/app-schema-test/pom.xml

```diff
@@ -163,6 +163,14 @@
             <version>2.0.0-1</version>
             <scope>test</scope>
         </dependency>
+        <!-- required for Web UI tests -->
+        <dependency>
+            <groupId>org.geoserver.web</groupId>
+            <artifactId>gs-web-demo</artifactId>
+            <version>${project.version}</version>
+            <classifier>tests</classifier>
+            <scope>test</scope>
+        </dependency>
```

Comment:
@ridethepenguin this branch fails to build (compile errors) in Maven as it has a different dependency model from Eclipse, which lumps all phases and classifiers together. I think you need test phase dependencies on gs-web-core tests and gs-web-demo main to get GeoServerWicketTestSupport and MapPreviewPage respectively. I got compilation working in Maven with this:

        <!-- required for Web UI tests -->
        <dependency>
            <groupId>org.geoserver.web</groupId>
            <artifactId>gs-web-demo</artifactId>
            <version>${project.version}</version>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>org.geoserver.web</groupId>
            <artifactId>gs-web-core</artifactId>
            <version>${project.version}</version>
            <classifier>tests</classifier>
            <scope>test</scope>
        </dependency>

Even with this I get a test failure:

Tests in error: 
  initializationError(org.geoserver.test.web.AbstractMapPreviewPageTest): Test class should have exactly one public constructor

Kind regards,
Ben.